src/vs/workbench/contrib/terminal/browser/terminalUri.ts

```diff
@@ -15,14 +21,17 @@ export function parseTerminalUri(resource: URI): ITerminalIdentifier {
 	return { workspaceId, instanceId: Number.parseInt(instanceId) };
 }
 
-export function getTerminalUri(workspaceId: string, instanceId: number, title?: string): URI {
+export function getTerminalUri(workspaceId: string, instanceId: number, title?: string, commandId?: string): URI {
+
 	return URI.from({
 		scheme: Schemas.vscodeTerminal,
 		path: `/${workspaceId}/${instanceId}`,
 		fragment: title || undefined,
+		query: commandId
```

Comment:
The query string should be `command=${commandId}`, just a guid or number after `?` isn't valid